consensus/core/src/core.rs

```diff
@@ -370,6 +384,43 @@ impl Core {
             }
         }
 
+        // TODO: produce the block for the clock_round. As the threshold clock can advance many rounds at once (ex
+        // because we synchronized a bulk of blocks) we can decide here whether we want to produce blocks per round
```

Comment:
I think we are reasonably sure we only want to build the latest blocks. Earlier blocks do not contribute to voting, and they can be close to GC or voting limit.